### PR TITLE
Taskomatic memory improvements

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Refactor the `UbuntuErrataManager` to consume less memory.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: Would require performance tests
- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/19718

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java [hunk 1]
```diff
@@ -1045,6 +1047,35 @@ public static Package lookupPackageByFilename(Channel channel,
...
+     * @param archLabel the architecture label
+     * @param version the package version
+     * @param release the package release
+     * @param epoch the package epoch (can be null)
+     * @param evrType the EVR type
+     *
+     * @return the list of matching packages
+     */
+    @SuppressWarnings("unchecked")
+    public static List<Package> lookupPackagesByNameArchEvr(Channel channel, String name, String archLabel,
```
Comment (lucidd) [inline java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java:1064]: same here

**Review by mcalmer — APPROVED** [review]
beside of @lucidd remarks looks good

**Review by rjmateus — APPROVED** [review]

File: java/code/src/com/redhat/rhn/domain/channel/Channel.hbm.xml [hunk 2]
```diff
@@ -569,6 +569,28 @@ PUBLIC "-//Hibernate/Hibernate Mapping DTD 3.0//EN"
     <return alias="p" class="com.redhat.rhn.domain.rhnpackage.Package" />
     </sql-query>
 
+    <sql-query name="Channel.packageByNameArchEvr">
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/domain/channel/Channel.hbm.xml]: addressed

File: java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java (hunk 1 above)
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java]: also addressed

**Pull request merged**